@xen-orchestra/web-core/lib/components/button/VtsButton.vue

```diff
@@ -0,0 +1,473 @@
+<!-- v4 -->
+<template>
+  <button :class="classNames" :disabled="busy || isDisabled" class="vts-button" type="button">
+    <VtsIcon :busy :icon="leftIcon" accent="current" class="icon" fixed-width />
+    <slot />
+    <VtsIcon :icon="rightIcon" accent="current" class="icon" fixed-width />
+  </button>
+</template>
+
+<script lang="ts" setup>
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import { useContext } from '@core/composables/context.composable'
+import { DisabledContext } from '@core/context'
+import type { ButtonLevel, ButtonSize } from '@core/types/button.type'
+import type { Color } from '@core/types/color.type'
+import { toVariants } from '@core/utils/to-variants.util'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = withDefaults(
+  defineProps<{
+    level: ButtonLevel
+    color: Color
+    size: ButtonSize
+    busy?: boolean
+    disabled?: boolean
+    leftIcon?: IconDefinition
+    rightIcon?: IconDefinition
+  }>(),
+  {
+    disabled: undefined,
+  }
+)
+
+const isDisabled = useContext(DisabledContext, () => props.disabled)
+
+const fontClasses = {
+  small: 'typo p3-medium',
+  medium: 'typo h6-medium',
+  large: 'typo h3-semi-bold',
+}
+
+const classNames = computed(() => [
+  fontClasses[props.size],
+  toVariants({
+    color: props.color,
+    level: props.level,
```

Comment:
If we update the props' names, we also need to update the classes here and in the CSS:
```suggestion
    accent: props.accent,
    variant: props.variant,
```